code/modules/antagonists/heretic/heretic_antag.dm

```diff
@@ -84,33 +84,99 @@
 	LAZYNULL(sac_targets)
 	return ..()
 
+/datum/antagonist/heretic/proc/get_icon_of_knowledge(datum/heretic_knowledge/knowledge)
+	if(ispath(knowledge))
+		//if the argument is a typepath, we need to make a dummy so we can access some of its properties
+		knowledge = new knowledge()
```

Comment:
And preferably the cached icons be generated in a way that doesn't require instantiating these types like this